state_transfer/transport/transport.go

```diff
@@ -28,6 +28,15 @@ type Transport interface {
 	Direct() bool
 	CreateServer(client.Client, endpoint.Endpoint) error
 	CreateClient(client.Client, endpoint.Endpoint) error
+	Options() *Options
+}
+
+type Options struct {
+	ProxyURL      string
```

Comment:
is Proxy a separate sub option? I wonder if making this its own type will make the Options extensible in the future?